### DISCUSSION TITLE
RelativeBinding

### DISCUSSION THREAD

**Discussion by hhgm** [AvaloniaUI/Avalonia#14319] [General]
Hi, I am creating a usercontrol which has a datacontext. This datacontext has a property which is a collection which I pass to the itemssource of a datagid. This works fine.
On the itemtemplate in the datagrid, I want a column which is a combobox, and the itemssource should be databound to a property on the datacontext of the usercontrol which contains a list of strings.
I have tried both the syntax $parent[class].property, as well as setting a relativesource binding.. nothing has been working so far

				`<DataGrid.Columns>`
					<DataGridTextColumn Header="Mode"
										Binding="{Binding Mode}"
										Width="120"/>
					<DataGridTemplateColumn Header="Template">
						<DataGridTemplateColumn.CellTemplate>
							<DataTemplate>
								<ComboBox
									ItemsSource="{Binding $parent[viewModels:DevicesViewModel].SelectedSatellite.Templates, Mode=OneWay, Converter={StaticResource PrefixListItemsConverter}, ConverterParameter=Templates/}"
									SelectedItem="{Binding ViewPath, Mode=TwoWay }"
									IsEnabled="{Binding IsConnected, Mode=OneWay}"
									Margin="2"/>
							</DataTemplate>
						</DataGridTemplateColumn.CellTemplate>
					</DataGridTemplateColumn>

					<DataGridTextColumn Header="Test" Binding="{Binding Path=DataContext, RelativeSource={RelativeSource AncestorType={x:Type Window}}}"/>

					<DataGridTemplateColumn Header="Decorations"
											Width="100">
						<DataGridTemplateColumn.CellTemplate>
							<DataTemplate>
								<ComboBox
									ItemsSource="{Binding $parent[models:SatelliteInfo].Decorations, Mode=OneWay}"
									SelectedItem="{Binding }"
									IsEnabled="{Binding IsConnected, Mode=OneWay}"
									Margin="2"/>
							</DataTemplate>
						</DataGridTemplateColumn.CellTemplate>
					</DataGridTemplateColumn>
				</DataGrid.Columns>
			</DataGrid>

all good ideas are welcome

**Comment by thevortexcloud** [comment]
This does not look right. You are binding to the _whole_ data context in your relative binding instead of a specific property. Your `$parent` binding also looks wrong as `$parent` will be a control and it's unlikely you have a VM directly in your hierachy. In the case of a DataGrid I think the parent is a row container?

If you press F12 to open dev tools it should make it easier to see what exactly you need to bind to.

**Comment by timunie** [comment]
1. I would give UserControl a name and by by `#NamedElement` syntax
2. If you use compiled bindings (I reconmend to do) you need to cast the values.

```xml
<ComboBox ItemsSoruce="{Binding #root.((vm:MainViewModel)DataContext).MyItems, Converter=...}" />
   [ ... ]
```

**Reply by hhgm** [reply]
Thanks a lot... I made it work by naming the user control and using that to access the datacontext
